apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useCaseActionsLogic/useCaseActionsLogic.tsx

```diff
@@ -57,38 +55,10 @@ export const useCaseActionsLogic = ({ workflowId, fullName }: IUseActions) => {
 
   // Avoid passing the onClick event to mutate
   const onMutateApproveEntity = useCallback(() => mutateApproveEntity(), [mutateApproveEntity]);
-  const childPendingWorkflowEvents = useMemo(
-    () =>
-      workflow?.childWorkflows?.flatMap(
-        childWorkflow => childWorkflow?.context?.pendingWorkflowEvents,
-      ) ?? [],
-    [workflow?.childWorkflows],
+  const { onMutateRevisionCase, pendingWorkflowEvents } = usePendingRevisionEvents(
```

Comment:
I understand the need to extract logic from `useCaseActionsLogic` but `usePendingRevisionEvents` should only return `pendingWorkflowEvents`. We can extract `onMutateRevisionCase` in some other way or at least achieve the same result via composition. (some hook consuming `usePendingRevisionEvents` etc.

Reply:
so what exactly do you suggest? removing the hook of pending revision from useCaseAction?